schematic/manifest/generator.py

```diff
@@ -1584,45 +1558,86 @@ def get_manifest(
                 # Update `additional_metadata` and generate manifest
                 manifest_url, manifest_df = self.get_manifest_with_annotations(annotations)
             
+            # Update df with existing manifest. Agnostic to output format
+            updated_df, out_of_schema_columns = self._update_dataframe_with_existing_df(empty_manifest_url=empty_manifest_url, existing_df=manifest_df)
+
             # determine the format of manifest that gets return 
-            result = self._handle_output_format_logic(output_format = output_format, output_path = output_path, sheet_url = sheet_url, empty_manifest_url=empty_manifest_url, dataframe = manifest_df)
+            result = self._handle_output_format_logic(output_format = output_format,
+                                                      output_path = output_path,
+                                                      sheet_url = sheet_url,
+                                                      empty_manifest_url=empty_manifest_url,
+                                                      dataframe = manifest_df,
+                                                      )
             return result
 
+    def _update_dataframe_with_existing_df(self, empty_manifest_url: str, existing_df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Handle scenario when existing manifest does not match new manifest template due to changes in the data model:
+            the sheet column header reflect the latest schema the existing manifest column-set may be outdated
+            ensure that, if missing, attributes from the latest schema are added to the column-set of the existing manifest so that the user can modify their data if needed
+            to comply with the latest schema.
+        Args:
+            empty_manifest_url (str): Google Sheet URL with an empty manifest with headers.
+            existing_df (Pd.DataFrame): df of an existing manifest
+
+        Returns:
+            updated_df (Pd.DataFrame): existing_df with new_columns added.
```

Comment:
Based on the code, it seems like we are returning both updated_df and out_of_schema_columns?

Reply:
Oop yep, ill update the returns.